### DISCUSSION TITLE
Flickering Mod

### DISCUSSION THREAD

**Discussion by angtang10** [riffnshred/nhl-led-scoreboard#288] [General]
Can you please help me with changing my raspberry pi zero w to this> --led-gpio-mapping=adafruit-hat-pwm

I know there are different commands like sudo and nano in putty, but I don't understand how to change this setting. Can I do it in Putty or should I pull out my SD card and change it in boot on the computer. Please help! This is like another language to me.

**Comment by pjockey** [comment]
If you are going to help remove the flicker you should perform the hardware upgrade. See links below.

https://github.com/hzeller/rpi-rgb-led-matrix#improving-flicker
https://github.com/hzeller/rpi-rgb-led-matrix#if-you-have-an-adafruit-hat-or-bonnet

Then follow the command line command in the set up instructions.

**Reply by angtang10** [reply]
Thanks for the reply. I have a jumper wire between those pins but don't know how to program the pi with --led-gpio-mapping=adafruit-hat-pwm. Do I make an empty file? Do I go into config.txt? I've had no luck giving a code like sudo --led-gpio-mapping=adafruit-hat-pwm.

**Comment by pjockey** [comment]
If you did the mod, then first try and see if the this command works for you.

$ cd nhl-led-scoreboard
sudo python3 src/main.py --led-gpio-mapping=adafruit-hat-pwm --led-brightness=60 --led-slowdown-gpio=2

**Reply by angtang10** [reply]
It went through, but made the flickering worse. Maybe I'll just live with the it haha. Appreciate your help.

**Reply by pjockey** [reply]
What size of board do you have?  Did you say you have a Pi 3b+? or a Zero W?

**Reply by angtang10** [reply]
I’ve got the 64x32 and the zero w. I’m definitely not a coder just really love hockey so tried to follow the instructions.

**Comment by riffnshred** [comment]
Hey bud, 
At the point where the software is, it's impossible to not have flickering. This is because it only has one core and with all the features we added, it takes everything it can to function and this will shot on the display. The anti-flicker mod should still help, and what @pjockey sent you should work, but you will still have some flickering. 

One thing you can try is to set everything Weather-related to false in the settings. this should at least reduce the flicker significantly.

Otherwise, best solution is using a Pi 3 and up

**Reply by angtang10** [reply]
Okay thank you. So I don’t run into any out of date problems again is there a particular model I should be getting? I just want what works the best not concerned with the price.

**Reply by riffnshred** [reply]
If you can find one, my go too is the 3A+. Be aware that if you are using a Bonnet (Small black board) and not the HAT (Big bleu board), you will have to Isolate the bottom of the it because it can touch the USB socket. You can use electric tape or kapton tape (2 layer of Kapton tape is what I did). 

I got 3 setup like that my self and will never change, so you are golden with that for the future.

**Reply by angtang10** [reply]
Thank you both very much! @pjockey

**Reply by pjockey** [reply]
Pi 3's are  good ones with the wifi built in or any of the Pi 4's. 

@riffnshred is more RAM better? or does this not a big factor? Any other instance it is.

**Reply by riffnshred** [reply]
The amount of Ram on the Pi Zero is more than enough to run this app. The only reason why the Pi Zero struggle so much is because its a single core and the rendering of the matrix itself take 60% of that core alone. With a multicore, the job is split and therefore no more bottleneck. More ram and arguably, more CPU power would not make a difference. With any model of the Pi 3 the software rolls at its peak performance.

**Reply by riffnshred** [reply]
The reason why I stay away from the Pi 4 for this project is because you have to power the raspberry pi with its own PSU. With a Pi 3, if you use Raspberry pi OS Lite and you don't plug anything in the USB port,  you can feed it through the HAT or Bonnet

**Reply by pjockey** [reply]
Thanks for the eduction. Never thought about using the Pi3A+ for any projects. Brings some new perspective.

**Comment by riffnshred** [comment]
No problem bud. And thanks for the help recently.
